src/mlpack/tests/test_catch_tools.hpp

```diff
@@ -21,23 +21,49 @@
 #define REQUIRE_RELATIVE_ERR(L, R, E) \
     REQUIRE(std::abs((R) - (L)) <= (E) * std::abs(R))
 
+template <typename MatType, typename ElemType = typename MatType::elem_type,
+          typename = std::enable_if_t<std::is_same<ElemType,
+              typename MatType::elem_type>::value>>
+class MatProxy {
+public:
+  explicit MatProxy(const MatType& mat) : mat(mat) { }
+  
+  const arma::Mat<ElemType> mat;
```

Comment:
You could also keep this a bit simpler by just accessing `mat` directly, up to you.